NuRadioReco/modules/LOFAR/beamformingDirectionFitter_LOFAR.py

```diff
@@ -89,40 +68,49 @@ def _direction_fit(self, station, ant_positions):
         dominant_pol = np.argmax(np.bincount(dominant_pol_traces))
         self.logger.debug(f"Dominant polarisation is {dominant_pol}")
 
-        # Collect the Efield traces - the e-field from the converter has eR as [0] component -> do +1 in index
-        fft_traces = np.array([trace.get_frequency_spectrum()[dominant_pol + 1]
+        # Collect the Efield traces
+        fft_traces = np.array([trace.get_frequency_spectrum()[dominant_pol]
                                for trace in station.get_electric_fields()])
 
         def negative_beamed_signal(direction):
             theta = direction[0]
             phi = direction[1]
-            direction_cartesian = hp.spherical_to_cartesian(theta, phi)
-
-            delays = geometric_delays(ant_positions, direction_cartesian)
-
-            out = beamformer(fft_traces, freq, delays)
-            timeseries = fft.freq2time(out, 200 * units.MHz)  # TODO: is this really necessary?
+            my_direction_cartesian = hp.spherical_to_cartesian(theta, phi)
+            my_out = mini_beamformer(fft_traces, freq, ant_positions, my_direction_cartesian)
+            timeseries = fft.freq2time(my_out, 200 * units.MHz)
 
             return -100 * np.max(timeseries ** 2)
 
         start_direction = np.array([station.get_parameter(stationParameters.zenith) / units.rad,
                                     station.get_parameter(stationParameters.azimuth) / units.rad])
-        fit_direction = fmin_powell(negative_beamed_signal, start_direction,
-                                    maxiter=self.__max_iter, xtol=1.0, disp=False)
+        self.logger.debug(f"Initial guess for fit routine is {start_direction}")
+
+        # TODO: adding Bounds changes result of fitting routine!
```

Comment:
Is this uncommented block still needed? It is a bit worrisome to have random ToDos in there.

Reply:
It is a bit complicated. Personally, I think this block should be here (uncommented of course), but this has a weird influence on the final result which I have not been able to figure out yet. Since we do not plan to use the beamforming fitting routine in the standard LOFAR pipeline, this has not been developed further. So calling it a TODO is probably not longer relevant.